src/utils/isNodeProcess.react-native.test.ts

```diff
@@ -0,0 +1,25 @@
+// @ts-nocheck
+/**
+ * Test in React Native Environment
+ */
+
+import { isNodeProcess } from './isNodeProcess'
+
+test('returns true when run in a react native environment', () => {
+  const originalProcess = global.process
+  global.process = {
+    env: {
+      NODE_ENV: 'development',
```

Comment:
Please, why do we have to simulate the development value here?

Reply:
here, my thought is that the NODE_ENV value doesn't have influence, so I set it to the default value in `React Native`.

Reply:
I think not having `global.process` explicitly at all brings us closer to testing the React Native logic in isolation.

Let's say we've got an issue in our `isNodeProcess()` that produces false positives. Now, since we also stub `global.process`, it may fall into the first condition that checks the `global.process` and produce a false positive. 

I think we shouldn't touch `global` in this test. What do you think?

Reply:
But if we don't touch `global.process`. `Object.prototype.toString.call(global.process) === '[object process]'` will be true, which should be false in `React Native`, and `navigator.product` will also not be evaluated. `isNodeProcess` will always return true. So I think maybe we should mock `global.process` and let `Object.prototype.toString.call(global.process)` should not be '[object process]'?

Reply:
Oh, you're totally right. Then this `global.process` assignment makes sense (still would move it to the `beforeAll` hook). Could you please also add a brief explanatory comment why we do that? (because the test itself runs in Node, we don't want it to return false positive). Thanks!

Reply:
I have moved mock in `beforeAll` and `afterAll` hook, and add some comments. Looks good to you?